scantool/scantool.c

```diff
@@ -640,6 +640,15 @@ l2_do_send(struct diag_l2_conn *d_conn, void *data, size_t len, void *handle)
 
 	msg.len = len;
 	msg.data = (uint8_t *)data;
+
+	if (d_conn->l2proto->diag_l2_protocol == DIAG_L2_PROT_VAG) {
```

Comment:
In your example, if 02 is the data, I presume one would do that with "sendreq 08 02" ? If so why even set msg.type then ? Can L2_vag not peek into data[] and figure out the same information ? I realize we won't ever be able to have a perfect abstraction for every protocol out there, but we can do the best we can.